### PR TITLE
Make player able to move to any inhabitants or their adjacent

### PR DESCRIPTION
**Pull request by 84634E1A607A**
**Related Issues**
Maybe there isn't one?

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

**Comment by 84634E1A607A** [comment]
I found myself that if I went extinct from two patches in a row, after entering the editor the second extinction disappeared (I still have population in that patch). Is it because #3460 ?

**Comment by hhyyrylainen** [comment]
> I found myself that if I went extinct from two patches in a row, after entering the editor the second extinction disappeared (I still have population in that patch). Is it because 

It shouldn't be related. More likely the external effects was not enough to overwhelm the population auto-evo calculated for you in that one patch you actually survived. This would be a pretty hard auto-evo interaction with gameplay design problem to solve.

File: src/microbe_stage/editor/PatchMapEditorComponent.cs [hunk 1]
```diff
@@ -149,49 +143,42 @@ private bool IsPatchMoveValid(Patch? patch)
...
-            return true;
-
         // If we are freebuilding, check if the target patch is connected by any means, then it is allowed
         if (Editor.FreeBuilding && CurrentPatch.GetAllConnectedPatches().Contains(patch))
             return true;
 
-        // Can't move if out of moves
-        if (!canStillMove)
-            return false;
+        // Can move to any patch that player species inhabits or their adjacent
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/PatchMapEditorComponent.cs:154]:
```suggestion
        // Can move to any patch that player species inhabits or is adjacent to such a patch
```

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
I didn't notice any issues in playtesting, I have one suggestion as to make one comment clearer.

**Review by hhyyrylainen — APPROVED** [review]
This is a small change and it's been open for a while, so I'll go ahead and commit my own suggestion and merge this.

**Pull request merged**